File: android/src/main/java/com/anji/appsp/sdk/model/SpRespUpdateModel.java
package com.anji.appsp.sdk.model;


import java.io.Serializable;

/**
 * @author chenhailong
 * Create Date:2020.7.2
 */


public class SpRespUpdateModel implements Serializable {
    //     apk download url
    public String url;
    //      if set true,we snap to H5 page to download apk,
    //      otherwise,we install apk internally
    public boolean isExternalUrl;
    //      check if show update dialog
    public boolean showUpdate;
    //     if set true,we do block user interaction,
    //     we exit application if we do not want to update,
    //     otherwise,user can cancel update schedule,
    public boolean mustUpdate;

    public int statusCode;
    //release note
    public String updateLog;

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public boolean isExternalUrl() {
        return isExternalUrl;
    }

    public void setExternalUrl(boolean externalUrl) {
        isExternalUrl = externalUrl;
    }

    public boolean isShowUpdate() {
        return showUpdate;
    }

    public void setShowUpdate(boolean showUpdate) {
        this.showUpdate = showUpdate;
    }

    public boolean isMustUpdate() {
        return mustUpdate;
    }

    public void setMustUpdate(boolean mustUpdate) {
        this.mustUpdate = mustUpdate;
    }

    public String getUpdateLog() {
        return updateLog;
    }

    public void setUpdateLog(String updateLog) {
        this.updateLog = updateLog;
    }

    public int getStatusCode() {
        return statusCode;
    }

    public void setStatusCode(int statusCode) {
        this.statusCode = statusCode;
    }

    @Override
    public String toString() {
        return "SpRespUpdateModel{" +
                "url='" + url + '\'' +
                ", isExternalUrl=" + isExternalUrl +
                ", showUpdate=" + showUpdate +
                ", mustUpdate=" + mustUpdate +
                ", statusCode=" + statusCode +
                ", updateLog='" + updateLog + '\'' +
                '}';
    }
}

File: lib/sp_resp_update_model.dart
class SpRespUpdateModel {
  bool isExternalUrl;
  bool mustUpdate;
  bool showUpdate;
  String code;
  String errorMsg;
  String updateLog;
  String url;

  SpRespUpdateModel({this.isExternalUrl,
    this.mustUpdate,
    this.showUpdate,
    this.code,
    this.errorMsg,
    this.updateLog,
    this.url});


  SpRespUpdateModel.fromJson(Map<String, dynamic> json) {
    isExternalUrl = json['isExternalUrl'];
    mustUpdate = json['mustUpdate'];
    showUpdate = json['showUpdate'];
    code = json['code'];
    errorMsg = json['errorMsg'];
    updateLog = json['updateLog'];
    url = json['url'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['isExternalUrl'] = this.isExternalUrl;
    data['mustUpdate'] = this.mustUpdate;
    data['showUpdate'] = this.showUpdate;
    data['code'] = this.code;
    data['errorMsg'] = this.errorMsg;
    data['updateLog'] = this.updateLog;
    data['url'] = this.url;
    return data;
  }


  @override
  String toString() {
    return 'SpRespUpdateModel{isExternalUrl: $isExternalUrl, mustUpdate: $mustUpdate, showUpdate: $showUpdate, code: $code, errorMsg: $errorMsg, updateLog: $updateLog, url: $url}';
  }


}
